terraform/interpolate.go

```diff
@@ -507,32 +500,37 @@ func (i *Interpolater) computeResourceMultiVariable(
 			continue
 		}
 
-		attr, ok := r.Primary.Attributes[v.Field]
-		if !ok {
-			// computed list attribute
-			_, ok := r.Primary.Attributes[v.Field+".#"]
-			if !ok {
-				continue
-			}
-			attr, err = i.interpolateListAttribute(v.Field, r.Primary.Attributes)
-			if err != nil {
-				return "", err
+		if singleAttr, ok := r.Primary.Attributes[v.Field]; ok {
+			if singleAttr == config.UnknownVariableValue {
+				return unknownVariable(), nil
 			}
+
+			values = append(values, singleAttr)
+			continue
 		}
 
-		if config.IsStringList(attr) {
-			for _, s := range config.StringList(attr).Slice() {
-				values = append(values, s)
-			}
+		// computed list attribute
+		_, ok = r.Primary.Attributes[v.Field+".#"]
+		if !ok {
 			continue
 		}
+		multiAttr, err := i.interpolateListAttribute(v.Field, r.Primary.Attributes)
+		if err != nil {
+			return emptyVariable(), err
+		}
 
-		// If any value is unknown, the whole thing is unknown
-		if attr == config.UnknownVariableValue {
-			return config.UnknownVariableValue, nil
+		if multiAttr.Value == unknownVariable() {
+			return emptyVariable(), nil
 		}
 
-		values = append(values, attr)
+		for _, element := range multiAttr.Value.([]ast.Variable) {
+			strVal := element.Value.(string)
+			if strVal == config.UnknownVariableValue {
+				return unknownVariable(), nil
+			}
```

Comment:
Just mentioning that here would be another place that #3449 might need to modify when we attack that.